src/Form/Form.js

```diff
@@ -895,10 +896,15 @@ class Form {
 
         // After autofill we check if form values match the data provided…
         const formValues = this.getValuesReadyForStorage();
+        const hasOnlyOneCredential =
+            (Boolean(formValues.credentials?.username) && !formValues.credentials?.password) ||
+            (Boolean(formValues.credentials?.password) && !formValues.credentials?.username);
         const areAllFormValuesKnown = Object.keys(formValues[dataType] || {}).every(
             (subtype) => formValues[dataType][subtype] === data[subtype],
         );
-        if (areAllFormValuesKnown) {
+        // If all form values are known, but we only have a single credntial field - then we want to prompt a partial save with username,
+        // So that in multi step forms (like reset-password), we can identify which username was picked, or complete a password save.
+        if (areAllFormValuesKnown && !hasOnlyOneCredential) {
```

Comment:
This change makes the submit call more "liberal" in a password step, but the JS side has no way of knowing what was filled before, and should be handled on the native side.

See: https://app.asana.com/0/72649045549333/1208836794233689/f